zmq/types/src/json_message_types.rs

```diff
@@ -0,0 +1,645 @@
+//! Objects for JSON serialization and deserialization in message bodies of
+//! the ZMQ pub/sub interface. Handles JSON for the following subscriptions:
+//! * `json-full-txpool_add` (`Vec<TxPoolAdd>`)
+//! * `json-minimal-txpool_add` (`Vec<TxPoolAddMin>`)
+//! * `json-full-chain_main` (`Vec<ChainMain>`)
+//! * `json-minimal-chain_main` (`ChainMainMin`)
+//! * `json-full-miner_data` (`MinerData`)
+use cuprate_types::hex::HexBytes;
+use serde::{Deserialize, Serialize};
+
+/// ZMQ `json-full-txpool_add` packets contain an array of `TxPoolAdd`.
+///
+/// Each `TxPoolAdd` object represents a new transaction in the mempool that was
+/// not previously seen in a block. Miner coinbase transactions *are not*
+/// included. `do-not-relay` transactions *are* included. Values are not
+/// republished during a re-org.
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+pub struct TxPoolAdd {
```

Comment:
Is this the same structure as here: https://github.com/Cuprate/cuprate/blob/b57ee2f4cf83dcb5fc63626a0cefe426d2b7a418/types/src/json/tx.rs#L28

Sorry for not noticing sooner if it is.

Reply:
They are similar, but their JSON layout, unfortunately, is completely different.

Reply:
Ughh why does monero have 2 JSON tx formats :sob: